### PR TITLE
fix image mapping function

### PR DESCRIPTION
**Pull request by zerlinazl**
New version correctly calculates M transformation matrix by taking 3 arbitrary xy points and mapping them to uv to get M, which is then applied on entire image. 
+ merry Christmas and happy holidays!

### CODE REVIEW

**Review by MyYo — CHANGES_REQUESTED** [review]
Thanks that looks good.

Can you explain why the previous transformation didn't work, while this one does work?
Worth adding a test that explains that.

**Comment by zerlinazl** [comment]
Here is an example of a test case that shows why the old function fails. I added it to the tester code for the new image_to_physical but included the code with a comment showing where the old function fails below.

I'm having a hard time articulating why exactly that happens. Here's my intuition: in the old function, we would do get_uv_from_xyz([bounding_box_x, bounding_box_y]). However, that meant the bounding box points would be treated the same as xy coordinates, which is not usually true, so the 'source' points wouldn't correspond to the 'dest' points and the calculated affine transformation wouldn't match the transformation defined by the uv mapping.

def test_old_image_physical_fails(self):
        # Create dummy plane
        uv = [[0,0],[100,0],[0,300]] # pix
        xyz = [[0,0,0],[1,0,0],[0,3,0]] # mm
        fp1 = FitPlane.from_template_centers(uv,xyz)

        # A second plane, mapping shifted over by 10px
        uv = [[10,0],[110,0],[10,300]] # pix
        xyz = [[0,0,0],[1,0,0],[0,3,0]] # mm
        fp2 = FitPlane.from_template_centers(uv,xyz)

        # Create a black image with one single white pixel at [50,50]
        np.random.seed(42)
        blank = np.zeros((300, 100, 3))
        blank[50,50] = np.array([255,255,255])

        # Transform the images 
        fp1_image = fp1.image_to_physical(blank, [0,2], [0,2], 0.01)
        fp2_image = fp2.image_to_physical(blank, [0,2], [0,2], 0.01)

        # Find the position of the white pixel in each transformed image and make sure it is in the correct place
        fp1_px = np.argwhere(np.all(fp1_image == np.array([255,255,255]), axis=-1)) # Should contain [50,50]
        fp2_px = np.argwhere(np.all(fp2_image == np.array([255,255,255]), axis=-1)) # Should contain [50,40]

        # First, check to make sure the white pixel is still in frame 
        assert len(fp1_px) > 0
        assert len(fp2_px) > 0    # The old image_to_physical function fails here, the white pixel disappeared

        self.assertAlmostEqual(fp1_px[0][0], fp2_px[0][0])
        self.assertAlmostEqual(fp1_px[0][1], fp2_px[0][1]+10)  # New function passes all the tests

File: plane/test_fit_plane.py [hunk 1]
```diff
@@ -157,6 +157,35 @@ def test_image_to_physical_output_size(self):
         self.assertAlmostEqual(random_image[10,10,1],mapped_image[10,10,1])
         self.assertAlmostEqual(random_image[50,50,2],mapped_image[50,50,2])
 
+    def test_image_physical(self):
```
Comment (MyYo) [inline plane/test_fit_plane.py:160]: Please rename test function:
test_image_physical --> test_image_to_physical_translations_uv

And rename function below
test_image_to_physical_translations --> test_image_to_physical_translations_xy

File: plane/test_fit_plane.py [hunk 2]
```diff
@@ -157,6 +157,35 @@ def test_image_to_physical_output_size(self):
...
+        # Create dummy plane with a random image
+        uv = [[0,0],[100,0],[0,300]] # pix
+        xyz = [[0,0,0],[1,0,0],[0,3,0]] # mm
+        # xyz = [[0,-1,0],[2,1,0],[0,3,0]] # mm
```
Comment (MyYo) [inline plane/test_fit_plane.py:164]: This line can be removed if not in use.

**Review by MyYo — CHANGES_REQUESTED** [review]
Thanks!
That example makes a lot of sense. 
I ask that you change tester function name in two places for clarity.
But otherwise, looks good!

Thanks for the clarification

**Review by MyYo — APPROVED** [review]
Looks good thanks!

**Pull request merged**